packages/web3/src/entitlement.ts

```diff
@@ -685,8 +693,27 @@ async function evaluateERC20Operation(
     )
 }
 
+async function evaluateCustomEntitledOperation(
+    operation: CheckOperation,
+    controller: AbortController,
+    provider: ethers.providers.StaticJsonRpcProvider,
+    linkedWallets: string[],
+): Promise<EntitledWalletOrZeroAddress> {
```

Comment:
this doesn't return `EntitledWalletOrZeroAddress`... it returns the first wallet in the linked wallets list. Also why did you choose to return the zero address instead of a boolean here?

Reply:
`EntitledWalletOrZeroAddress` is a bit of a strange type - it is intended to be an address that is either valid or the zero address - the zero address is the placeholder that indicates there is no entitled wallet for this permission in the user's set of wallets, i.e. they are not entitled. In the first case I return the first wallet because the entitled call is checking all of the wallets and so we don't know which of the wallets in particular is entitled, although multiple wallets may also be entitled. But I can definitely refactor this so that we test each wallet independently and only return a wallet that passed on the contract.

Reply:
Definitely don't think we should call these individually, that seems like a waste. Do we ever use the EntitledWallet return value anywhere?

Reply:
Yes, I think that Evan uses it in `joinTown` and also `mintMembershipTransaction` on the towns client in harmony, but I'm not 100% sure what it's doing.